sklearn/linear_model/_logistic.py

```diff
@@ -696,28 +316,34 @@ def _logistic_regression_path(
     # multinomial case this is not necessary.
     if multi_class == "ovr":
         w0 = np.zeros(n_features + int(fit_intercept), dtype=X.dtype)
-        mask_classes = np.array([-1, 1])
         mask = y == pos_class
         y_bin = np.ones(y.shape, dtype=X.dtype)
-        y_bin[~mask] = -1.0
-        # for compute_class_weight
+        if solver in ["lbfgs", "newton-cg"]:
+            mask_classes = np.array([0, 1])
+            y_bin[~mask] = 0.0
```

Comment:
Note that the new log-loss/HalfBinomialLoss uses `y in [0, 1]` instead of `y in [-1, 1]`.

Side note: Additional benefit is that y could now also be any value in the interval [0, 1], i.e. helps solving #17411.